script_parameters.yml

```diff
@@ -0,0 +1,238 @@
+# scripts).  See the Script parameter reference linked below for a reference
+# containing all the available parameter types, and what variables are made
+# available to your script.  If you are passing in string parameters from user
+# input or other untrusted sources, we advise using escaped versions of the
+# variables to prevent injection attacks.
+#
+# Links:
+# * Workflows documentation (requires Platform login): https://civis.zendesk.com/hc/en-us/articles/115004172983-Workflows-Basics
+# * Script parameter reference (requires Platform login): https://civis.zendesk.com/hc/en-us/articles/360001579592-Configuring-Scripts-via-the-Civis-API
+# * Mistral domain specific language reference: https://docs.openstack.org/mistral/latest/user/wf_lang_v2.html
+# * YAML syntax: https://docs.ansible.com/ansible/latest/reference_appendices/YAMLSyntax.html
+
+version: '2.0'  # you always need this key to specify version 2 of the mistral DSL
+workflow:
+  tasks:
+    sql_script:
+      action: civis.scripts.sql
+      input:
+        # SQL scripts always require the IDs of the remote host and the credential
+        # you are using to connect to it.
+        remote_host_id: 0  # get it from https://api.civisanalytics.com/databases
+        credential_id: 0   # get it from https://platform.civisanalytics.com/credentials
+        params:  # This is not an exhaustive list of param types - see the script parameter reference for more
+          - name: my_integer
+            type: integer
+          - name: my_float
+            type: float
+          - name: my_string
+            type: string
+          - name: my_custom_credential
+            type: credential_custom
+        arguments:
+          my_integer: 1234
+          my_float: 0.1234
+          my_string: this is a string
+          my_custom_credential: 0  # ID of the credential
+        sql: |
+          SELECT 1;
+
+          -- Variables enclosed in double curly brackets will be replaced with the parameter values.
+          -- my_integer: {{my_integer}}
+          -- my_float: {{my_float}}
+
+          -- To prevent SQL injection attacks, you can get escaped versions of string parameters.
+          -- {{my_string.literal}} wraps the string in single quotes, changing single quotes in the string to double quotes.
+          -- {{my_string.identifier}} is in a format that is safe to be used as a Redshift identifier.
+
+          -- my_custom_credential: {{my_custom_credential.username}} {{my_custom_credential.password}}
+      on-success:
+        - container_script
+
+    container_script:
+      action: civis.scripts.container
+      input:
+        docker_image_name: civisanalytics/datascience-python
+        docker_image_tag: latest
+        required_resources:
+          cpu: 256
+          memory: 512
+          disk_space: 1
+        params:
+          # Optionally, you can specify the parameter values in the `params` list
```

Comment:
The wording here might be a little confusing.  I imagine I would use this by skimming to the particular type of script I want to put in my workflow and I might be a little thrown off by the word 'optionally' if I skipped straight to this section

Reply:
Hmm yeah, I see what you mean.  I guess there's a tradeoff here between accommodating people who will read the whole file, and people who just want an example for the script type they're trying to use.  I can duplicate more of the comments across each of the script types, but I'm worried that the redundancy will annoy people reading the whole file.  What do you think?

Reply:
I think this is a good point. Maybe some of these things can go in the commented section at the top? It actually wasn't that hard for me to find any info that I thought I might want by skimming around, so this is a really minor issue anyway.